dbms/src/Storages/DeltaMerge/SSTFilesToDTFilesOutputStream.cpp

```diff
@@ -319,12 +319,11 @@ std::vector<ExternalDTFileInfo> SSTFilesToDTFilesOutputStream<ChildStream>::outp
 template <typename ChildStream>
 void SSTFilesToDTFilesOutputStream<ChildStream>::cancel()
 {
-    // Try a lightweight cleanup the file generated by this stream (marking them able to be GC-ed).
     for (auto & file : ingest_files)
     {
         try
         {
-            file->enableGC();
+            file->remove(context.getFileProvider());
```

Comment:
The file's size may already be added into `StableDiskDelegator` in `SSTFilesToDTFilesOutputStream::writeSuffix`. And how to remove the size of this file?